README.md

```diff
@@ -58,7 +113,7 @@ the download key (and possibly the collection key) into the asynchronous functio
 status can be updated.
 
 Arguments:
-- `pk`: the unique identifier for a collection of downloads - this will typically be a user PK
+- `username`: the unique identifier for a collection of downloads - this will typically be a user name
```

Comment:
```suggestion
- `username`: the unique identifier for a collection of downloads - this will typically be a username
```
We might want to refine this a little. At the moment we are allowing it to be a int, and assuming it is a user PK, which will be backwards compatible with our usage. Happy to leave things as they are for this PR though, as I think such changes are largely independent of websockets.

Reply:
72be20e, we are allowing it to be user obj, int or string and function `get_username()`  tries to match user from db to given argument and get his `username`. 
How about adding there also searching user by given string and add general else for case if all our 'match ifs' fails?

Reply:
I guess I'm trying to think about if the download should always be associated with a single user, or if there are cases where you wouldn't want that (e.g. some kind of "global" downloads all users could access). That was the original reason this argument was changed from expecting a user object to a more general "key".

We could call the argument `user`, which then doesn't really specify whether it is an object, username string or PK, allowing all.

To be honest though, most reasonable use-case for calling `init_download` is going to be in a view where you have a `request` object, and therefore you have access to `request.user`, so I'd say allowing only an object or PK is fine, and disallowing a username. I would say go back to only allowing a user object, but that would break all current implementations that pass `request.user.pk`.

Reply:
I removed 'sting' variant and now exception will be raised if we pass something else than user object or int (user.pk) 2f2c3df